src/workerd/api/r2-bucket.c++

```diff
@@ -841,7 +841,7 @@ kj::Array<R2Bucket::Etag> parseConditionalEtagHeader(
     return parseConditionalEtagHeader(condHeader.slice(nextComma + 1), kj::mv(etagAccumulator));
   } else if (leadingCommaRequired) {
   // Did not find a leading comma, and we expected a leading comma before any further etags
-    KJ_FAIL_REQUIRE("Comma was expected to separate etags");
+    JSG_FAIL_REQUIRE("Comma was expected to separate etags");
```

Comment:
These will need the error types added, e.g.
```suggestion
    JSG_FAIL_REQUIRE(Error, "Comma was expected to separate etags");
```

Reply:
thanks, done, other parts of the R2 binding seem to often prefer TypeError for stuff like this, but just Error seems more appropriate here.